pkg/types/azure/validation/platform.go

```diff
@@ -80,6 +80,9 @@ func ValidatePlatform(p *azure.Platform, publish types.PublishingStrategy, fldPa
 		if p.ARMEndpoint != "" {
 			allErrs = append(allErrs, field.Required(fldPath.Child("armEndpoint"), fmt.Sprintf("ARM endpoint must not be set when the cloud name is %s", cloud)))
 		}
+		if p.RHCOSImage != "" {
+			allErrs = append(allErrs, field.Required(fldPath.Child("RHCOSImage"), fmt.Sprintf("RHCOS Image must not be set when the cloud name is %s", cloud)))
```

Comment:
I forgot to mention you also need to check the platform_test.go to see if the existing tests are impacted by this change and also might need to add tests.